src/main/java/fr/openent/competences/helper/DateHelper.java

```diff
@@ -0,0 +1,44 @@
+package fr.openent.competences.helper;
+
+import fr.openent.competences.constants.DateFormat;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DateHelper {
+    private static final Logger log = LoggerFactory.getLogger(DateHelper.class);
+
+    private DateHelper() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static Date parse(String dateString, String format) {
+        Date date = null;
+        try {
+            date = new SimpleDateFormat(format).parse(dateString);
+        } catch (ParseException e) {
+            log.error(String.format("[Competence@%s::parseDate] Error when casting date. %s",
```

Comment:
add LoggingGuards: `ifErrorEnabled`

Reply:
What you want me to do right there ? can you give me an exemple ?

Reply:
```
if(log.isErrorEnabled()){
   log.error("{} text has some expensive method called and may not even do get logged because of configured loglevel", parse(data));
}
```

A logging guards saves you cpu load whenever something will get logged and calls a method to parse or gather data/information. As the logging might is disabled or the log level does not match the selected log level its cpu load which is lost for nothing:

https://stackoverflow.com/a/12953090/2739334

Reply:
In general I recommend doing it, LogLevel.ERROR probably is an exception to it (because otherwise it would be disabled or FATAL), but it can not hurt.

Reply:
Ok ok ! thank you for all those advice !